src/components/ClusterNodes/NodeDetails.tsx

```diff
@@ -0,0 +1,765 @@
+    BreadCrumb,
+    ButtonWithLoader,
+    copyToClipboard,
+    Pagination,
+    Progressing,
+    showError,
+    useBreadcrumb,
+} from '../common'
+import { ReactComponent as Info } from '../../assets/icons/ic-info-filled.svg'
+import { ReactComponent as Error } from '../../assets/icons/ic-error-exclamation.svg'
+import { ReactComponent as AlertTriangle } from '../../assets/icons/ic-alert-triangle.svg'
+import { ReactComponent as Cpu } from '../../assets/icons/ic-cpu.svg'
+import { ReactComponent as Memory } from '../../assets/icons/ic-memory.svg'
+import { ReactComponent as Storage } from '../../assets/icons/ic-storage.svg'
+import { ReactComponent as Edit } from '../../assets/icons/ic-pencil.svg'
+import PageHeader from '../common/header/PageHeader'
+import { useParams } from 'react-router'
+import { ReactComponent as Clipboard } from '../../assets/icons/ic-copy.svg'
+import Tippy from '@tippyjs/react'
+import { ReactComponent as Success } from '../../assets/icons/appstatus/healthy.svg'
+import CodeEditor from '../CodeEditor/CodeEditor'
+import YAML from 'yaml'
+import { getNodeCapacity, updateNodeManifest } from './clusterNodes.service'
+import {
+    NodeDetail,
+    NodeDetailResponse,
+    PodType,
+    ResourceDetail,
+    TEXT_COLOR_CLASS,
+    UpdateNodeRequestBody,
+} from './types'
+import { toast } from 'react-toastify'
+import { ReactComponent as Sort } from '../../assets/icons/ic-sort-arrow.svg'
+import { OrderBy } from '../app/list/types'
+
+export default function NodeDetails() {
+    const [loader, setLoader] = useState(false)
+    const [apiInProgress, setApiInProgress] = useState(false)
+    const [isReviewState, setIsReviewStates] = useState(false)
+    const [selectedTabIndex, setSelectedTabIndex] = useState(0)
+    const [selectedSubTabIndex, setSelectedSubTabIndex] = useState(0)
+    const [nodeDetail, setNodeDetail] = useState<NodeDetail>(null)
+    const { clusterId, nodeName } = useParams<{ clusterId: string; nodeName: string }>()
+    const [copied, setCopied] = useState(false)
+    const [manifest, setManifest] = useState('')
+    const [modifiedManifest, setModifiedManifest] = useState('')
+    const [cpuData, setCpuData] = useState<ResourceDetail>(null)
+    const [memoryData, setMemoryData] = useState<ResourceDetail>(null)
+    const [sortedPodList, setSortedPodList] = useState<PodType[]>(null)
+    const [podListOffset, setPodListOffset] = useState(0)
+    const [sortByColumnName, setSortByColumnName] = useState<string>('name')
+    const [sortOrder, setSortOrder] = useState<string>(OrderBy.ASC)
+    const pageSize = 10
+
+    useEffect(() => {
+        setLoader(true)
+        getNodeCapacity(clusterId, nodeName)
+            .then((response: NodeDetailResponse) => {
+                if (response.result) {
+                    response.result.pods.sort((a, b) => a['name'].localeCompare(b['name']))
+                    setSortedPodList(response.result.pods.sort((a, b) => a['name'].localeCompare(b['name'])))
```

Comment:
and why are we doing the same sorting twice?

Reply:
Done